Sources/RxCBCentral/core/RxPeripheralImpl.swift

```diff
@@ -326,7 +328,7 @@ class RxPeripheralImpl: NSObject, RxPeripheral, CBPeripheralDelegate {
     private let didReadRSSISubject = PublishSubject<Int>()
     private let didDiscoverServicesSubject = PublishSubject<([CBService], Error?)>()
     private let didDiscoverCharacteristicsSubject = PublishSubject<([CBCharacteristic], Error?)>()
-    private let didUpdateValueForCharacteristicSubject = PublishSubject<(CBCharacteristic, Error?)>()
+    internal let didUpdateValueForCharacteristicSubject = PublishSubject<(CBCharacteristic, Error?)>()
```

Comment:
I don't think we want to expose this state to other classes if we can help it.

The right way to approach this would be to follow a similar pattern I worked on for the `ConnectionManager` and `RxCentralDelegate`:

```swift
public class ConnectionManager: ConnectionManagerType {
    
// note this this is an internal-only initializer that lets you pass in an object 
// that conforms to the RxCentralDelegate delegate protocol, allowing you to mock and test.
    init(...
         delegate: RxCentralDelegate, 
         ...) {
        ...
        centralManager.delegate = delegate
        
        setupDelegate()
    }
   
   // Public init with no ability to pass-in a delegate - created internally.
   public convenience init(...) {
        self.init(..., delegate: RxCentralDelegateImpl(), ...)
    }
```

Reply:
Thanks for the feedback! I changed the (internal) init to receive the subject I need for testing, it does look a lot better now. Later we can extend this to the others to make this entire class easier to test :)